tests/test_object_states.py

```diff
@@ -850,14 +1049,84 @@ def test_draped():
     with pytest.raises(NotImplementedError):
         carpet.states[Draped].set_value(breakfast_table, False)
 
+
 @og_test
-def test_covered():
-    breakfast_table = og.sim.scene.object_registry("name", "breakfast_table")
+def test_filled():
+    stockpot = og.sim.scene.object_registry("name", "stockpot")
 
-    place_obj_on_floor_plane(breakfast_table)
+    systems = (
+        get_system("water"),
+        get_system("raspberry"),
+        get_system("diced_apple"),
+    )
+    for system in systems:
+        stockpot.set_position_orientation(position=np.ones(3) * 50.0, orientation=[0, 0, 0, 1.0])
+        place_obj_on_floor_plane(stockpot)
+        for _ in range(5):
+            og.sim.step()
 
-    for _ in range(5):
+        assert stockpot.states[Filled].set_value(system, True)
+
+        for _ in range(5):
+            og.sim.step()
+
+        assert stockpot.states[Filled].get_value(system)
+        stockpot.states[Filled].set_value(system, False)
+
+        for _ in range(5):
+            og.sim.step()
+        assert not stockpot.states[Filled].get_value(system)
+
+        system.remove_all_particles()
+
+
+@og_test
+def test_contains():
+    stockpot = og.sim.scene.object_registry("name", "stockpot")
+
+    systems = (
+        get_system("water"),
+        get_system("stain"),
+        get_system("raspberry"),
+        get_system("diced_apple"),
+    )
+    for system in systems:
+        stockpot.set_position_orientation(position=np.ones(3) * 50.0, orientation=[0, 0, 0, 1.0])
+        place_obj_on_floor_plane(stockpot)
+        for _ in range(5):
+            og.sim.step()
+
+        # Sample single particle
+        if is_physical_particle_system(system_name=system.name):
+            system.generate_particles(positions=[np.array([0, 0, stockpot.aabb[1][2] + system.particle_radius * 1.01])])
+            assert not stockpot.states[Contains].get_value(system)
+        else:
+            if system.get_group_name(stockpot) not in system.groups:
+                system.create_attachment_group(stockpot)
+            system.generate_group_particles(
+                group=system.get_group_name(stockpot),
+                positions=np.array([np.array([0, 0, stockpot.aabb[1][2] - 0.1])]),
+                link_prim_paths=[stockpot.root_link.prim_path],
+            )
+
+        for _ in range(10):
+            og.sim.step()
+
+        assert stockpot.states[Contains].get_value(system)
+
+        # Remove all particles and make sure contains returns False
+        system.remove_all_particles()
         og.sim.step()
+        assert not stockpot.states[Contains].get_value(system)
+
+        # Cannot set Contains state
+        with pytest.raises(NotImplementedError):
+            stockpot.states[Contains].set_value(system, True)
+
+
+@og_test
+def test_covered():
```

Comment:
shall we add covered(cloth, particles) to this test as well? Maybe after you implement the new feature of sampling the physical particles on cloth?

Reply:
tbd, later